src/components/Search/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from "react";
+import styles from "./Search.module.scss";
+
+type TProps = {
+  searchInput: string;
+  updateSearchInput: (value: string) => void;
+};
+
+class Search extends Component<TProps> {
+  input: React.RefObject<HTMLInputElement>;
+
+  constructor(props: TProps) {
+    super(props);
+    this.input = React.createRef();
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    const { updateSearchInput } = this.props;
+    const value: string = this.input.current!.value.trim();
```

Comment:
```suggestion
    const value: string = this.input.current?.value.trim(); 
```
your ref object is null initially, it would be better if we check for it firstly